srcts/src/components/sidebar.ts

```diff
@@ -242,6 +266,307 @@ class Sidebar {
+      passive: false,
+    });
+    document.addEventListener("touchmove", this._onResizeMove.bind(this), {
+      passive: false,
+    });
+    document.addEventListener("touchend", this._onResizeEnd.bind(this));
+
+    // Keyboard events for accessibility
+    handle.addEventListener("keydown", this._onResizeKeyDown.bind(this));
+
+    // Prevent text selection during resize
+    handle.addEventListener("selectstart", (e) => e.preventDefault());
+
+    window.addEventListener(
+      "resize",
+      whenChangedCallback(
+        () => this._getWindowSize(),
+        () => this._updateResizeAvailability()
+      )
+    );
+  }
+
+  /**
+   * Check if the sidebar should be resizable in the current state.
+   * @private
+   * @returns {boolean}
+   */
+  private _shouldEnableResize(): boolean {
+    const isDesktop = this._getWindowSize() === "desktop";
+    const notTransitioning = !this.layout.container.classList.contains(
+      Sidebar.classes.TRANSITIONING
+    );
+    const notClosed = !this.isClosed;
+
+    return (
+      // Allow resizing only when the sidebar...
+      isDesktop && notTransitioning && notClosed
+    );
+  }
+
+  /**
+   * Handle resize start (mouse/touch down).
+   * @private
+   * @param {MouseEvent | TouchEvent} event
+   */
+  private _onResizeStart(event: MouseEvent | TouchEvent): void {
+    if (!this._shouldEnableResize()) return;
+
+    event.preventDefault();
+
+    const clientX =
+      "touches" in event ? event.touches[0].clientX : event.clientX;
+
+    this.resizeState.isResizing = true;
+    this.resizeState.startX = clientX;
+    this.resizeState.startWidth = this._getCurrentSidebarWidth();
+
+    // Disable transitions during resize for smooth interaction
+    this.layout.container.style.setProperty("--_transition-duration", "0ms");
+    this.layout.container.classList.add(Sidebar.classes.RESIZING);
+
+    // Change mouse cursor and prevent text selection
+    document.body.style.cursor = this._isRightSidebar()
+      ? "ew-resize"
+      : "ew-resize";
+    document.body.style.userSelect = "none";
+
+    this._dispatchResizeEvent("start", this.resizeState.startWidth);
+  }
+
+  /**
+   * Handle resize move (mouse/touch move).
+   * @private
+   * @param {MouseEvent | TouchEvent} event
+   */
+  private _onResizeMove(event: MouseEvent | TouchEvent): void {
+    if (!this.resizeState.isResizing) return;
+
+    event.preventDefault();
+
+    const clientX =
+      "touches" in event ? event.touches[0].clientX : event.clientX;
+    const deltaX = clientX - this.resizeState.startX;
+
+    // Calculate new width based on sidebar position
+    const isRight = this._isRightSidebar();
+    const newWidth = isRight
+      ? this.resizeState.startWidth - deltaX
+      : this.resizeState.startWidth + deltaX;
+
+    // Constrain within bounds
+    const constrainedWidth = Math.max(
+      this.resizeState.minWidth,
+      Math.min(this.resizeState.maxWidth(), newWidth)
+    );
```

Comment:
I feel on the fence on this, but this calculation occurs twice, so maybe worth a helper method?